### DISCUSSION TITLE
predefined Metadata definition

### DISCUSSION THREAD

**Discussion by Dau-Duc-Hai** [pimcore/pimcore#12387] [Q&A / Support] [answered]
error appears when click on "predefined Metadata definition" or "grid options" 
Please have look;
The message as bellow:  
Pimcore\Model\Asset\Metadata\Loader\DataLoader::build(): Argument #1 ($name) must be of type string, null given, called in /var/www/html/pimcore/vendor/pimcore/pimcore/models/Metadata/Predefined.php on line 368

![image](https://user-images.githubusercontent.com/51470610/173016283-b3f6ff3d-0031-4774-9418-2eb7152ca025.png)

**Comment by kingjia90** [comment, marked as answer]
Can you double check if everything is correctly set here?
![image](https://user-images.githubusercontent.com/6014195/173667074-12900af0-dadd-4df2-abf6-339c564aef5a.png)
In case, also click on an empty cell of each row and press `Enter` to trigger a save, so everything is "refreshed".

if you cannot access that page either, please check in the `settings_store` with `scope = 'pimcore_predefined_asset_metadata'` if there's any missing/corruption in the `data` column
OR 
check configuration files inside `var/config/predefined-asset-metadata/`

**Reply by Dau-Duc-Hai** [reply]
Thank you Ji,
for reply the situation as follow:
Predefined metadata: cannot accessible
 settings_store: structure is good but this table is empty (no record) 
var/config/predefined-asset-metadata/ have no error
the "grid configuration" have the same error 
![image](https://user-images.githubusercontent.com/51470610/173812735-41d1045a-63ca-4f92-a612-eb049f65a3c2.png)
but if I change line 368 to:  $instance = $loader->build($this->type ?? "xx" ); I can access to "grid configuration" but predefined metadata still error

Please advice 
Thank you

**Reply by kingjia90** [reply]
Thank you for the detailed reply!
This narrows down the origin of the problem, could be that one or more of the files in `var/config/predefined-asset-metadata/` must be "corrupted". Make sure the yml looks like this.

```
pimcore:
    assets:
        metadata:
            predefined:
                definitions:
                    ....
                    type: anything-but-null
                    ....
```
or if you have the debugger, you can check if the $this for each entry is containing all the info, especially for the `type`
![image](https://user-images.githubusercontent.com/6014195/173818767-da1fc787-485e-44d5-97fd-632364adb859.png)

The workaround `$instance = $loader->build($this->type ?? "xx" );` could bypass this error, but it indicates other underlying problems, because `type` shouldn't be `null`.

**Reply by Dau-Duc-Hai** [reply]
Thank you very much JI,

You right, there is null type in var/config/predefined-asset-metadata/
I delete it and it working now.

Thank you for support.
Best regards, 

HD
